library/src/main/java/com/datatorrent/lib/expressions/ExpressionEvaluator.java

```diff
@@ -0,0 +1,440 @@
+   * The expression provided here is not expected to contain a return statement.
+   * Expression should be a single line expression as per JLS 7 as given in:
+   * <a>https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html</a>
+   *
+   * <p>
+   *  Example is as follows:
+   *  Let the expression be <expression>
+   *  After compiling and resolving the placeholder, it becomes <compiled_expression>
+   *
+   *  The execute method of Expression interface would look like this:
+   *  <return_type> execute(Object... obj)
+   *  {
+   *    return ((<return_type>)(<compiled_expression>))
+   *  }
+   * </p>
+   *
+   * @param expression  String expression which has return statement in it.
+   * @param returnType  Type of class which will be returned by execute method of Expression interface.
+   * @param <O>         Parameterized return class type of expression after evaluation.
+   * @return            Object of type Expression interface having compiled and executable expression.
+   */
+  public <O> Expression<O> createExecutableExpression(String expression, Class<?> returnType)
+  {
+    return createExecutableCode(expression, returnType, false);
+  }
+
+  /**
+   * This is a private which will be called by public method createExecutableExpression & createExecutableFunction.
+   * This method is the main method which converts the given expression into java expression format and compiles it
+   * using Janino.
+   *
+   * The method works in following steps:
+   * <ol>
+   *   <li>Resolve object/variable placeholder to respective types and replace them in expression</li>
+   *   <li>Generate final code to go into execute method of Expression class. Here the final code is generated based
+   *       on whether the containsReturnStatement is true OR false. If its true, the code is added as it is
+   *       If false, it'll added along with return statement.</li>
+   *   <li>Compile the expression using Janino library</li>
+   *   <li>Return an object of type Expression which contains compiled expression.</li>
+   * </ol>
+   *
+   * @param expression              String expression that needs to be compiled
+   * @param returnType              Return type of the expression.
+   * @param containsReturnStatement Whether the expression contains return statement OR not.
+   * @param <O>                     Parameterized return class type of expression after evaluation.
+   * @return                        Object of type Expression interface having compiled and executable expression.
+   */
+  @SuppressWarnings({"unchecked"})
+  private <O> Expression<O> createExecutableCode(String expression, Class<?> returnType,
+      boolean containsReturnStatement)
+  {
+    if (this.placeholderClassMapping.size() <= 0) {
+      throw new RuntimeException("Mapping needs to be provided.");
+    }
+
+    Pattern entry = Pattern.compile(variablePlaceholderPattern);
+    Matcher matcher = entry.matcher(expression);
```

Comment:
Considering this will be run in some operator, when the expression variable is encountered, it'll anyway throw exception. 
Do you see any advantage in adding this check before that?